src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Argument
+{
+    private ?int $mode = null;
+
+    /**
+     * Represents a console command <argument> definition.
+     *
+     * If unset, the `name`, `mode` and `default` values might be inferred from the parameter definition.
```

Comment:
Changing the **name** is useful, especially when you want to change from `myOptionName` to `my-option-name`.

This **mode** is not available in the constructor parameters. No need to mention it in this comment.

For the **default** value, what benefit in setting it in the attribute rather than in the signature?